evm/src/cpu/decode.rs

```diff
@@ -28,122 +20,103 @@ use Availability::{All, Kernel, User};
+    (0x0b, 0, false, COL_MAP.is_signextend),
+    (0x10, 0, false, COL_MAP.is_lt),
+    (0x11, 0, false, COL_MAP.is_gt),
+    (0x12, 0, false, COL_MAP.is_slt),
+    (0x13, 0, false, COL_MAP.is_sgt),
+    (0x14, 0, false, COL_MAP.is_eq),
+    (0x15, 0, false, COL_MAP.is_iszero),
+    (0x16, 0, false, COL_MAP.is_and),
+    (0x17, 0, false, COL_MAP.is_or),
+    (0x18, 0, false, COL_MAP.is_xor),
+    (0x19, 0, false, COL_MAP.is_not),
+    (0x1a, 0, false, COL_MAP.is_byte),
+    (0x1b, 0, false, COL_MAP.is_shl),
+    (0x1c, 0, false, COL_MAP.is_shr),
+    (0x1d, 0, false, COL_MAP.is_sar),
+    (0x20, 0, false, COL_MAP.is_keccak256),
+    (0x30, 0, false, COL_MAP.is_address),
+    (0x31, 0, false, COL_MAP.is_balance),
+    (0x32, 0, false, COL_MAP.is_origin),
+    (0x33, 0, false, COL_MAP.is_caller),
+    (0x34, 0, false, COL_MAP.is_callvalue),
+    (0x35, 0, false, COL_MAP.is_calldataload),
+    (0x36, 0, false, COL_MAP.is_calldatasize),
+    (0x37, 0, false, COL_MAP.is_calldatacopy),
+    (0x38, 0, false, COL_MAP.is_codesize),
+    (0x39, 0, false, COL_MAP.is_codecopy),
+    (0x3a, 0, false, COL_MAP.is_gasprice),
+    (0x3b, 0, false, COL_MAP.is_extcodesize),
+    (0x3c, 0, false, COL_MAP.is_extcodecopy),
+    (0x3d, 0, false, COL_MAP.is_returndatasize),
+    (0x3e, 0, false, COL_MAP.is_returndatacopy),
+    (0x3f, 0, false, COL_MAP.is_extcodehash),
+    (0x40, 0, false, COL_MAP.is_blockhash),
+    (0x41, 0, false, COL_MAP.is_coinbase),
+    (0x42, 0, false, COL_MAP.is_timestamp),
+    (0x43, 0, false, COL_MAP.is_number),
+    (0x44, 0, false, COL_MAP.is_difficulty),
+    (0x45, 0, false, COL_MAP.is_gaslimit),
+    (0x46, 0, false, COL_MAP.is_chainid),
+    (0x47, 0, false, COL_MAP.is_selfbalance),
+    (0x48, 0, false, COL_MAP.is_basefee),
+    (0x49, 0, true, COL_MAP.is_prover_input),
+    (0x50, 0, false, COL_MAP.is_pop),
+    (0x51, 0, false, COL_MAP.is_mload),
+    (0x52, 0, false, COL_MAP.is_mstore),
+    (0x53, 0, false, COL_MAP.is_mstore8),
+    (0x54, 0, false, COL_MAP.is_sload),
+    (0x55, 0, false, COL_MAP.is_sstore),
+    (0x56, 0, false, COL_MAP.is_jump),
+    (0x57, 0, false, COL_MAP.is_jumpi),
+    (0x58, 0, false, COL_MAP.is_pc),
+    (0x59, 0, false, COL_MAP.is_msize),
+    (0x5a, 0, false, COL_MAP.is_gas),
+    (0x5b, 0, false, COL_MAP.is_jumpdest),
+    (0x5c, 0, true, COL_MAP.is_get_state_root),
+    (0x5d, 0, true, COL_MAP.is_set_state_root),
+    (0x5e, 0, true, COL_MAP.is_get_receipt_root),
+    (0x5f, 0, true, COL_MAP.is_set_receipt_root),
+    (0x60, 5, false, COL_MAP.is_push), // 0x60-0x7f
+    (0x80, 4, false, COL_MAP.is_dup),  // 0x80-0x8f
+    (0x90, 4, false, COL_MAP.is_swap), // 0x90-0x9f
+    (0xa0, 0, false, COL_MAP.is_log0),
+    (0xa1, 0, false, COL_MAP.is_log1),
+    (0xa2, 0, false, COL_MAP.is_log2),
+    (0xa3, 0, false, COL_MAP.is_log3),
+    (0xa4, 0, false, COL_MAP.is_log4),
     // Opcode 0xa5 is PANIC when Kernel. Make the proof unverifiable by giving it no flag to decode to.
```

Comment:
Just to double check, does this strategy for `PANIC` still work? It looks like now `handle_invalid` would panic, causing an infinite loop. I suppose it's unsatisfiable since the CPU can never end the loop and reach `halt`?

Reply:
Yeah, exactly. It’s a bit roundabout, but `PANIC` still works.